android/src/main/kotlin/com/onegini/mobile/sdk/flutter/handlers/CustomRegistrationHandler.kt

```diff
@@ -0,0 +1,49 @@
+package com.onegini.mobile.sdk.flutter.handlers
+
+import com.onegini.mobile.sdk.flutter.providers.CustomRegistrationAction
+import java.lang.Exception
+
+class CustomRegistrationHandler {
+    fun actionRespondSuccess(
+        idProvider: String?,
+        customRegistrationActions: ArrayList<CustomRegistrationAction>,
+        token: String?) {
+        val action = getCustomRegistrationAction(idProvider, customRegistrationActions)
+
+        when {
+            action != null ->  action.returnSuccess(token)
+            else -> return // TODO throw platform exception no provider found with that id once error rework is merged
+         }
+    }
+
+    fun actionRespondError(
+        idProvider: String?,
+        customRegistrationActions: ArrayList<CustomRegistrationAction>,
+        error: String?) {
+        if (error == null) {
+            // TODO Thow platform exception
+            return
+        }
+
+        val action = getCustomRegistrationAction(idProvider, customRegistrationActions)
+
+        when {
+            action != null ->  action.returnError(Exception(error))
+            else -> return // TODO throw platform exception
+        }
+    }
+
+    private fun getCustomRegistrationAction(idProvider: String?, customRegistrationActions: ArrayList<CustomRegistrationAction>): CustomRegistrationAction? {
```

Comment:
This function name is too long

Reply:
I think it's good, it's better to have longer, descriptive name than shorter than can be misunderstood.

Reply:
@PiotrKlis Sorry, I didnt mean the function name, I meant the line.